docs/blog/_posts/2018-10-05-10th-dotty-milestone-release.md

```diff
@@ -0,0 +1,288 @@
+compiler supporting them.
+
+Dotty is the project name for technologies that are considered for inclusion in Scala 3. Scala has
+pioneered the fusion of object-oriented and functional programming in a typed setting. Scala 3 will
+be a big step towards realizing the full potential of these ideas. Its main objectives are to
+
+- become more opinionated by promoting programming idioms we found to work well,
+- simplify where possible,
+- eliminate inconsistencies and surprising behaviors,
+- build on strong foundations to ensure the design hangs well together,
+- consolidate language constructs to improve the language’s consistency, safety, ergonomics, and
+  performance.
+
+You can learn more about Dotty on our [website](https://dotty.epfl.ch).
+
+<!--more-->
+
+This is our 10th scheduled release according to our
+[6-week release schedule](https://dotty.epfl.ch/docs/usage/version-numbers.html).
+
+## What’s new in the 0.10.0-RC1 technology preview?
+
+### Java 9+
+
+Dotty now supports the latest versions of Java. Try it out in our REPL!
+
+```scala
+scala> val ls = java.util.List.of(1, 2, 3)
+val ls: java.util.List[Int] = [1, 2, 3]
+```
+
+### Type-level programming: Match Types
+
+We introduces a new form of types called match types. A match type reduces to one of a number of
+right hand sides, depending on a scrutinee type. E.g:
+
+```scala
+type Elem[X] = X match {
+  case String => Char
+  case Array[t] => t
+  case Iterable[t] => t
+}
+```
+
+This defines a type that, depending on the scrutinee type X, can reduce to one of its right hand
+sides. For instance,
+
+```scala
+Elem[String]       =:=  Char
+Elem[Array[Int]]   =:=  Int
+Elem[List[Float]]  =:=  Float
+Elem[Nil]          =:=  Nothing
+```
+
+Here `=:=` is understood to mean that left and right hand sides are mutually subtypes of each other.
+
+For more information, visit the [Match Types](http://dotty.epfl.ch/docs/reference/match-types.html)
+section of our documentation.
+
+### Documentation in the REPL
+
+The previous release added documentation support for the IDE. Users can now query the documentation
+of sources previously compiled by Dotty within the REPL:
+
+```scala
+scala> /** An object */ object O { /** A def */ def foo = 0 }
+// defined object O
+
+scala> :doc O
+/** An object */
+
+scala> :doc O.foo
+/** A def */
+```
+
+### Tail-recursive methods can now be polymorphic
+
+Previously, a tail recursive call would be optimised only if the type arguments of the method
+did not change at call site. E.g.
+
+```scala
+@tailrec def loop[T](x: T): Int = {
+  ...
+  loop[Int](1)
+}
+```
+
+```shell
+   loop[Int](1)
+   ^^^^^^^^^^^^
+  Cannot rewrite recursive call: it changes type arguments on a polymorphic recursive call
+```
+
+We remove this restriction which has been a leaking implementation detail.
+
+### Name-based `unapplySeq` is now supported
+
+Similarly to the `unapply` method, an extractor object can now define a `def unapplySeq(x: T): U`
```

Comment:
I would note that this matches what Scala 2.13.0-M5 does, and add a link to http://dotty.epfl.ch/docs/reference/changed/pattern-matching.html